experimenter/evaluate_pipeline_new.py

```diff
@@ -0,0 +1,282 @@
+import itertools as it
+import json
+import os
+
+from typing import Any, List, Tuple
+from uuid import UUID
+
+from d3m.metadata.pipeline import Pipeline
+from d3m import cli
+
+from experimenter.databases.d3m_mtl import D3MMtLDB
+from data_preparation_pipelines import K_FOLD_TABULAR_SPLIT_PIPELINE_ID, SCORING_PIPELINE_ID
+
+def save_pipeline_run_to_d3m_db(pipeline_run_path: str):
+    """ 
+    Saves a pipeline run document to the d3m database.
+
+    Parameters
+    ----------
+    pipeline_run_path : path_like str
+        path to pipeline_run document
+
+    Returns:
+    ----------
+    TODO
+
+    Raises:
+    ----------
+    TODO
+    """
+    d3m_db = D3MMtLDB()
+    return D3MMtLDB().save_pipeline_run(pipeline_run_path)
+
+def evaluate_pipeline_on_problem(pipeline_path: str,
+    problem_path: str,
+    input_path: str,
+    data_random_seed: int):
+    """ 
+    Evaluate pipeline on problem.
+    A less verbose form of running d3m's runtime cli 'evaluate' command.
+    See 'evaluate_pipeline_via_d3m_cli' for more options for running 
+    the 'evaluate' command.
+
+    Parameters
+    ----------
+    pipeline_path : path_like str
+        path to pipeline doc
+    problem_path : path_like str 
+        path to problem doc
+    input_path : path_like str
+        path to input full data
+    data_random_seed : int   
+        random seed to be used for data preparation
+
+    Returns:
+    ----------
+    None
+
+    Raises:
+    ---------------------------------
+    OSError
+        when a file cannot be opened
+    """
+    output_run_path = []
+
+    with open(pipeline_path, 'r') as pipeline:
+        output_run_path.append(pipeline['properties']['digest'])
+    with open(problem_path, 'r') as problem:
+        output_run_path.append(problem['properties']['digest'])
+    with open(input_path, 'r') as input_f:
+        output_run_path.append(input_f['properties']['digest'])
+
+    output_run_path = '_'.join(output_run_path) + '.json'
+
+    execute_pipeline_via_d3m_cli(pipeline=pipeline_path, problem=problem_path,
+        input=input_path, output_run=output_run_path,
+        data_random_seed=data_random_seed)
+
+    save_pipeline_run_to_d3m_db(output_run_path)
+
+def evaluate_pipeline_via_d3m_cli(pipeline: str,
+    problem: str,
+    input: str,
+    output_run: str,
+    data_random_seed: int,
+    data_params: List[Tuple[str,Any]] = None,
+    data_pipeline: str = K_FOLD_TABULAR_SPLIT_PIPELINE_ID,
+    scoring_pipeline: str = SCORING_PIPELINE_ID,
+    input_run: str = None,
+    metric: str = None,
+    scoring_params: List[Tuple[str,Any]] = None,
+    scores: str = None,
+    scoring_random_seed: int = None,
+    data_split_file: str = None):
```

Comment:
What's the point of all these arguments if they are not going to be used?

Reply:
There is no point as of now, but I didn't know if they would be used in the future. They are simply all parameters from the `runtime` command. I'll remove then because we are not using them.